pycaret/internal/helper.py

```diff
@@ -0,0 +1,677 @@
+                       'dimension', 'feature', 'feature_all'],
+        'catboost  ': ['auc', 'pr', 'confusion_matrix', 'error', 'class_report', 'boundary', 'learning', 'vc',
+                       'dimension', 'feature', 'feature_all']
+    }
+    return plot_list_dict[model_name]
+
+
+def get_plot_list_clustering(model_name):
+    plot_list_dict = {
+        'kmeans ': ['cluster', 'tsne', 'elbow', 'silhouette', 'distance', 'distribution'],
+        'ap ': ['cluster', 'tsne', 'distance', 'distribution'],
+        'meanshift ': ['cluster', 'tsne', 'distance', 'distribution'],
+        'sc ': ['cluster', 'tsne', 'elbow', 'distribution'],
+        'hclust ': ['cluster', 'tsne', 'elbow', 'distribution'],
+        'dbscan ': ['cluster', 'tsne', 'distribution'],
+        'optics ': ['cluster', 'tsne', 'distribution'],
+        'birch ': ['cluster', 'tsne', 'elbow', 'silhouette', 'distribution'],
+        'kmodes ': ['cluster', 'tsne', 'elbow', 'silhouette', 'distribution']
+    }
+    return plot_list_dict[model_name]
+
+
+def get_plot_list_anomaly_detection(model_name):
+    plot_list_dict = {
+        'abod ': ['tsne'],
+        'cluster ': ['tsne'],
+        'cof ': ['tsne'],
+        'histogram ': ['tsne'],
+        'knn ': ['tsne'],
+        'lof ': ['tsne'],
+        'svm ': ['tsne'],
+        'pca ': ['tsne'],
+        'mcd ': ['tsne'],
+        'sod ': ['tsne'],
+        'sos ': ['tsne']
+    }
+    return plot_list_dict[model_name]
+
+
+def get_plot_list_nlp(model_name):
+    plot_list_dict = {
+        'lda ': ['frequency', 'distribution', 'bigram', 'trigram', 'sentiment', 'tsne', 'topic_distribution',
+                 'wordcloud'],
+        'lsi ': ['frequency', 'distribution', 'bigram', 'trigram', 'sentiment', 'tsne', 'topic_distribution',
+                 'wordcloud'],
+        'hdp ': ['frequency', 'distribution', 'bigram', 'trigram', 'sentiment', 'tsne', 'topic_distribution',
+                 'wordcloud'],
+        'rp ': ['frequency', 'distribution', 'bigram', 'trigram', 'sentiment', 'tsne', 'topic_distribution',
+                'wordcloud'],
+        'nmf ': ['frequency', 'distribution', 'bigram', 'trigram', 'sentiment', 'tsne', 'topic_distribution',
+                 'wordcloud']
+    }
+    return plot_list_dict[model_name]
+
+
+def get_report_name(report_type, model_type):
+    report_name = ''
+    if model_type == 'regression':
+        report_name = get_report_name_regression(report_type)
+    elif model_type == 'classification':
+        report_name = get_report_name_classification(report_type)
+    elif model_type == 'clustering':
+        report_name = get_report_name_clustering(report_type)
+    elif model_type == 'anomaly_detection':
+        report_name == get_report_name_anomaly_detection(report_type)
+    elif model_type == 'nlp':
+        report_name = get_report_name_nlp(report_type)
+    return report_name
+
+
+# Specific to each model type
+def get_report_name_regression(report_type):
```

Comment:
Should be using the built-in model names

Reply:
We are removing these methods and instead we will just call it model report

Reply:
Since maintaing these methods in the long run can be an overhead as pointed by Antonio